File: MyApplication/app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {
// buoc 1: xac dinh ca phan tu se dieu khien va khai bao dong thuoc tinh cua lop

    private Button btnExit,btnhello;
    private EditText editName;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
        //buoc 2:tham chieu
        btnExit = (Button) findViewById(R.id.btn_Exit);
        btnhello = (Button) findViewById(R.id.btn_Hello);
        editName = (EditText) findViewById(R.id.edt_Name);

        btnhello.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String name = edtName.getText
            }
        });

    }
}

File: OurSharedPreference/app/src/main/java/com/example/oursharedpreference/MainActivity.kt
package com.example.oursharedpreference

import android.annotation.SuppressLint
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import com.example.oursharedpreference.ui.theme.OurSharedPreferenceTheme
import org.w3c.dom.Text

class MainActivity : ComponentActivity() {
    //khaibao thuoc tinh cua 1 lop
    private lateinit var edtPhone: EditText
    private lateinit var btnSave: Button
    private lateinit var btbLoad: Button
    private lateinit var txtInfo: TextView

    @SuppressLint("MissingInflatedId")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //tham chieu ptu view thuc te tren giao dien
        edtPhone = findViewById(R.id.edt_phone)
        btnSave = findViewById(R.id.btn_save)
        btbLoad = findViewById(R.id.btn_load)
        txtInfo = findViewById(R.id.txt_info)

        btnSave.setOnClickListener {
            //lay ra gtr nhap vao tu edittext
            val phone = edtPhone.text.toString()
            // luu gtr nay vao Sharedpreferences
            val sharedPreferences = getSharedPreferences("HIEU", MODE_PRIVATE)
            val editor = sharedPreferences.edit()
            editor.putString("PHONE", phone)
            editor.apply()
        }

        btbLoad.setOnClickListener {
            val sharedPreferences = getSharedPreferences("HIEU", MODE_PRIVATE)
            val phone = sharedPreferences.getString("PHONE", "")
            txtInfo.text = phone
        }

    }
}
